Activiti-5522 upgrade issue in 8.1.3

ShreasiK opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 The commit message follows our guidelines
     - 👷‍♂️ Tests for the changes have been added (for bug fixes / features)
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.
-->

<!--
(check one with "x") one of the following
-->

**What kind of change does this PR introduce?**

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

## Description

<!--
You can also link to an open issue here
-->

- Issue Link: https://hyland.atlassian.net/browse/ACTIVITI-5522

<!--
If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
-->

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

ShreasiK committed:
Corrected the drop index syntax

ShreasiK committed:
Added 'DROP INDEX ACT_IDX_IDENT_LNK_TASK' in missing drop files & removed unwanted drop query from upgrade files & added CREATE INDEX ACT_IDX_IDENT_LNK_TASK for oracle create file

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Activiti/Activiti?pullRequest=5114) <br/>All committers have signed the CLA.

balsarori commented on activiti-core/activiti-engine/src/main/resources/org/activiti/db/upgrade/activiti.mssql.upgradestep.810.to.813.engine.sql:
```diff
@@ -1,6 +1,7 @@
-update ACT_GE_PROPERTY set VALUE_ = '8.1.3' where NAME_ = 'schema.version';
-
 drop index ACT_RU_IDENTITYLINK.ACT_IDX_IDENT_LNK_TASK;
+
```
Let's check if it exists before dropping
`IF EXISTS (SELECT name FROM sysindexes WHERE name = 'ACT_IDX_IDENT_LNK_TASK') drop index ACT_RU_IDENTITYLINK.ACT_IDX_IDENT_LNK_TASK;`

ShreasiK replied:
Fixed it

ShreasiK committed:
Added 813.to.814 upgrade files, modified 810.to.813 files accordingly

balsarori reviewed (changes requested):
- All create schema files should have schema.version 8.1.4 see https://github.com/Activiti/Activiti/blob/8.1.x/activiti-core/activiti-engine/src/main/resources/org/activiti/db/create/activiti.mssql.create.engine.sql#L9
- ProcessEngine.VERSION should be 8.1.4 see https://github.com/Activiti/Activiti/blob/8.1.x/activiti-core/activiti-engine/src/main/java/org/activiti/engine/ProcessEngine.java#L46
- Add previous 8.1.3 to ACTIVITI_VERSIONS see https://github.com/Activiti/Activiti/blob/8.1.x/activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/db/DbSqlSession.java#L150

ShreasiK committed:
updated schema.version to 8.1.4 in all create schema files.
updated ProcessEngine.VERSION to 8.1.4
added previous version (8.1.3) to ACTIVITI_VERSIONS

ShreasiK committed:
updated schema.history to 8.1.4 in all create schema files

ShreasiK committed:
Reverted changes of schema.history = 8.1.4 in all create schema files

arpanhyland commented on activiti-core/activiti-engine/src/main/resources/org/activiti/db/create/activiti.db2.create.engine.sql:
```diff
@@ -6,7 +6,7 @@ create table ACT_GE_PROPERTY (
+values ('schema.version', '8.1.4', 1);
 
 insert into ACT_GE_PROPERTY
 values ('schema.history', 'create(8.1.3)', 1);
```
Need to change this also for all create scripts.

ShreasiK replied:
Fixed it

ShreasiK committed:
Updated schema.history to create(8.1.4) in all create schema files

balsarori reviewed (approved):
LGTM
